lib/utils/flattenNestedSelectorsForRule.cjs

```diff
@@ -3,27 +3,45 @@
 'use strict';
 
 const resolveNestedSelector = require('postcss-resolve-nested-selector');
+const selectorParser = require('postcss-selector-parser');
 const getRuleSelector = require('./getRuleSelector.cjs');
 const isStandardSyntaxSelector = require('./isStandardSyntaxSelector.cjs');
 const parseSelector = require('./parseSelector.cjs');
 
 /**
  * @typedef {import('postcss-selector-parser').Selector} Selector
  * @typedef {import('postcss-selector-parser').Root} Root
+ * @typedef {{selector: Selector, resolvedSelectors: Root, nested: boolean}} FlattenedSelector
  *
  * @param {import('postcss').Rule} rule
  * @param {import('stylelint').PostcssResult} result
- * @returns {Array<{selector: Selector, resolvedSelectors: Root}>}
+ * @returns {Array<FlattenedSelector>}
  */
 function flattenNestedSelectorsForRule(rule, result) {
 	const ownAST = parseSelector(getRuleSelector(rule), result, rule);
 
 	if (!ownAST) return [];
 
+	/** @type {Array<FlattenedSelector>} */
 	const flattenedSelectors = [];
 
 	for (const selectorAST of ownAST.nodes) {
-		const resolvedSelectors = resolveNestedSelector(selectorAST.toString(), rule);
+		const selector = selectorAST.toString();
+
+		const resolvedSelectors = resolveNestedSelector(selector, rule);
+
+		if (resolvedSelectors.length === 1 && resolvedSelectors[0] === selector) {
+			flattenedSelectors.push({
+				selector: selectorAST,
+				resolvedSelectors: selectorParser.root({
+					value: '',
+					nodes: [selectorAST],
+				}),
+				nested: false,
+			});
+
+			continue;
+		}
 
 		for (const resolvedSelector of resolvedSelectors) {
 			if (!isStandardSyntaxSelector(resolvedSelector)) return [];
```

Comment:
question

When the `length === 1` shouldn't `isStandardSyntaxSelector` be called too?